Junit/src/IPL.java

```diff
@@ -0,0 +1,364 @@
+
+        List<Player> listOfPlayers = new ArrayList<Player>();
+
+        try{
+
+            BufferedReader reader = new BufferedReader(new FileReader("./src/IPL_2021_data.csv"));
+
+            // for passing the header
+            reader.readLine();
+            String line = "";
+            while((line = reader.readLine()) != null){
+                String[] playerArray = line.split(",");
+                listOfPlayers.add(new Player(playerArray[0],playerArray[1],playerArray[2],Integer.valueOf(playerArray[3]),Integer.valueOf(playerArray[4]),Double.valueOf(playerArray[5]),Double.valueOf(playerArray[6]),Integer.valueOf(playerArray[7])));
+            }
+
+            reader.close();
+        }
+        catch(Exception e){
+            System.out.println("File not found "+ e);
+        }
+
+        return listOfPlayers;
+    }
+
+    // for fixing the match between team and stadium and storing in a csv file
+    public static void fixMatch(){
+        final int totalTeams = 8;
+        String[] listOfTeams = new String[totalTeams];
+        String[] homeTeamPlace = new String[totalTeams];
+
+        listOfTeams[0] =  "CSK";
+        listOfTeams[1] =  "PBKS";
+        listOfTeams[2] =  "RCB";
+        listOfTeams[3] =  "SRH";
+        listOfTeams[4] =  "MI";
+        listOfTeams[5] =  "RR";
+        listOfTeams[6] =  "DC";
+        listOfTeams[7] =  "KKR";
+
+        homeTeamPlace[0] = "HOME CSK";
+        homeTeamPlace[1] = "HOME PBKS";
+        homeTeamPlace[2] = "HOME RCB";
+        homeTeamPlace[3] = "HOME SRH";
+        homeTeamPlace[4] = "HOME MI";
+        homeTeamPlace[5] = "HOME RR";
+        homeTeamPlace[6] = "HOME DC";
+        homeTeamPlace[7] = "HOME KKR";
+
+
+        ArrayList<String> matchList = new ArrayList<String>();
+        for(int i=0;i<totalTeams;i++){
+            for(int j=i+1;j<totalTeams;j++){
+                matchList.add(listOfTeams[i] + "," + listOfTeams[j] + "," + homeTeamPlace[i] + "\n");
+                matchList.add(listOfTeams[i] + "," + listOfTeams[j] + "," + homeTeamPlace[j] + "\n");
+            }
+        }
+
+        Collections.shuffle(matchList);
+
+
+
+        try{
+            FileWriter matchWriter = new FileWriter("match.csv");
+            for (String s : matchList) {
+                matchWriter.write(s);
+            }
+            matchWriter.close();
+        }
+        catch(Exception e){
+            System.out.println(e);
+        }
+
+
+    }
+
+    // finding bowler who took 40 wickets.
+    public static void searchBowlerAtleastFortyWickets(List<Player> listOfPlayers, String team){
+
+        for(Player p: listOfPlayers){
+            if(p.getWicket() >= 40 && p.getTeam().equals(team) && p.getRole().equals("BOWLER")){
+                System.out.println(p.getName());
+            }
+        }
+    }
+
+    // searching a player by it's first name last name or full name
+    public static void searchPlayer(List<Player> listOfPlayers,String query){
```

Comment:
1. should have been part of the team class
2. this does not handle all the cases, you are not doing a case insensitive search, hence it might fail for certain use cases.
3. using the contains method would have been a more graceful solution